### PR TITLE
Add support for DHCP CNI plugin IPAM options and set the DHCP host-name option properly

### LINKED ISSUE #1103: Supply container hostname in CNI_ARGS
### What is the problem you're trying to solve

When running containers with `ipam: { "type": "dhcp" }`   , the dhcp request have a empty hostname.
Makes life painful when using dhcp to find out on DHCP server side which container have that lease.

### Describe the solution you'd like

the "dhcp" type already have a mechanism to send information in the DHCP request:

> provide (dict, optional): Options providing to DHCP server when acquire leases.
> option (string, optional): String or number representation of the DHCP option.
> value (string, optional): String representation of the value. Directly sent to server.
> fromArg (string, optional): Get value from CNI_ARGS by given argument name.

(https://www.cni.dev/plugins/current/ipam/dhcp/)

Thus supplying the container hostname in a given key under CNI_ARGS would solve the issue.

### Additional context

Just for information, podman supplies the following CNI_ARGS on my tests: `IgnoreUnknown=1;K8S_POD_NAME=elastic_wilbur`

I can use the following configuration for my ipam:
```
...
"type": "dhcp",
"provide": [
  {
    "option": "host-name",
    "fromArg": "K8S_POD_NAME"
  }
]
...
```

Also, If there's no option supplied, the value doesn't get overriden (on my tests, CNI v1.1.1) . For example:
```
...
"type": "dhcp",
"provide": [
  {
    "option": "host-name",
    "value": "foo"
  },
  {
    "option": "host-name",
    "fromArg": "K8S_POD_NAME"
  }
]
...
```
Will sent the DHCP option "host-name" with value `foo` when `K8S_POD_NAME` is not present on CNI_ARGS.
This way one can have a default value for a given option (useful for vendor-class-identifier).
So anything as key could be used .. (i.e.: NERDCTL_HOSTNAME, etc.). Logic of which key goes where would stay in the ipam config.

### PR DESCRIPTION
**Pull request by apostasie**
Fixes #1103.

cc @AkihiroSuda @theodiem

### CODE REVIEW

File: docs/cni.md [hunk 1]
```diff
@@ -90,10 +90,33 @@ an easier way is to use DHCP to assign the IP:
...
+    --ipam-driver dhcp \
+    --ipam-opt 'vendor-class-identifier={"type": "provide", "value": "Hey! Its me!"}' \
+    my-dhcp-net
+```
+
 ## Custom networks
 
 You can also customize your CNI network by providing configuration files.
-For example you have one configuration file(`/etc/cni/net.d/10-mynet.conf`)
+For exemple you have one configuration file(`/etc/cni/net.d/10-mynet.conf`)
```
Comment (AkihiroSuda) [inline docs/cni.md:119]: ?

File: pkg/netutil/cni_plugin_unix.go [hunk 2]
```diff
@@ -18,7 +18,9 @@
 
 package netutil
 
-import "github.com/containerd/nerdctl/v2/pkg/rootlessutil"
+import (
+	"github.com/containerd/nerdctl/v2/pkg/rootlessutil"
+)
```
Comment (AkihiroSuda) [inline pkg/netutil/cni_plugin_unix.go:23]: ?

Comment (apostasie) [reply pkg/netutil/cni_plugin_unix.go]: Sorry. Pushed too fast. Both typo are fixed.

**Comment by apostasie** [comment]
Rebased.
The cni plugins sha failure is now gone.

PTAL.

Still some (random?) CI failure - unless I am responsible for these, it seems to me these are just flaky. It is really annoying.
I want these problems gone. Will have a look at them once done with image inspect.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**